src/main/java/jenkins/security/plugins/ldap/LDAPConfiguration.java

```diff
@@ -100,6 +100,11 @@
      */
     private final String server;
 
+    /**
+     * whether to verify ldaps sever certificate? default is false
+     */
+    private boolean sslVerify;
```

Comment:
This still defaults to `false` when it should default to `true`

Set the field to `true` in the `@DataboundConstructor`, the `@DataboundSetter`s will be called after the constructor and set the value to `false` if it was configured as such in the form field. old `JCasC` configs should then also still work.
You also need to set the default value in the form.

Next you also need to make sure that old configurations that are loaded from disk gets the default `true` value. The common way of solving that would be to make this field an object `Boolean` instead so that you can distinguish between a configured and non configured value. and then in the `readResolve` method set the field to `true` if it is `null`.
See https://wiki.jenkins.io/display/JENKINS/Hint+on+retaining+backward+compatibility for a bit more hints on how it works.

Then you should/could also add a test to the backwards compatibility. I am somewhat sure there already are some test for it so you can just add a `assertTrue(configuration.isSslVerify())` to it.

Reply:
Because the @DataboundSetters will be called after the constructor, the constructor cannot get the value configured in the form field.
This field is referenced in inferRootDN() . And the constructor calls inferRootDN() to infer the root DN when rootDN is blank and it is not allowed to be blank.  
So  calling the new signature with the default value for the new field. The old constructor signature is kept but deprecated.
Please help review the new changes.